### DISCUSSION TITLE
Google Pay is available on this device.

### DISCUSSION THREAD

**Discussion by Gopal6696** [flutter-stripe/flutter_stripe#805] [Q&A]
meta-data android:name="com.google.android.gms.wallet.api.enabled" android:value="true  added in manifest
implementation 'com.google.android.gms:play-services-wallet:19.1.0' added in build.gradle

Future<void> startGooglePay(String amount) async {
  try {
    final response = await createPaymentIntent(amount);
    final clientSecret = response['client_secret'];
    await Stripe.instance.initGooglePay(const GooglePayInitParams(
        testEnv: true, merchantName: "Test", countryCode: 'us'));
    const IsGooglePaySupportedParams(testEnv: true);
    await Stripe.instance.presentGooglePay(
      PresentGooglePayParams(clientSecret: clientSecret),
    );
  } catch (e) {
    print('Gpay Error: $e');
  }
}

createPaymentIntent(String amount) async {
  try {
    Map<String, dynamic> body = {
      'amount': amount.toString(),
      'currency': 'usd'
    };
    var response = await http.post(
        Uri.parse('https://api.stripe.com/v1/payment_intents'),
        body: body,
        headers: {
          'Authorization': 'Bearer $stripeSecretKey',
          'Content-Type': 'application/x-www-form-urlencoded'
        });
    return jsonDecode(response.body);
  } catch (err) {
    print(err.toString());
  }

Error Details
 Gpay Error: StripeException(error: LocalizedErrorMessage(code: FailureCode.Failed, localizedMessage: presentForPaymentIntent() may only be called when Google Pay is available on this device., message: presentForPaymentIntent() may only be called when Google Pay is available on this device., stripeErrorCode: null, declineCode: null, type: null)
[New Text Document (2).txt](https://github.com/flutter-stripe/flutter_stripe/files/8965044/New.Text.Document.2.txt)
)

**Comment by remonh87** [comment]
Did you also comply to: 
https://developers.google.com/pay/api/android/guides/setup  
https://stripe.com/docs/google-pay?platform=android  ?

This is an issue coming from the Stripe sdk and normally means the config is wrong in some area. It also helps to ask stripe for support.

**Reply by Gopal6696** [reply]
i had completed this steps.

still same error came.
i also tried to build "https://github.com/flutter-stripe/flutter_stripe" this example app.
that also give same error.

minsdk:21
targetsdk:32
compilesdk:32

**Reply by Gopal6696** [reply]
@remonh87

**Reply by remonh87** [reply]
I really think you need to configure sth on the Stripe or google side. This message is what we receive from Stripe itself and we just forward it.

**Reply by remonh87** [reply]
also did you configure a payment card on the phone itself?

**Reply by Gopal6696** [reply]
@remonh87 card payment is working

**Reply by remonh87** [reply]
No for google pay you need to configure a payment card that is linked to your google account on the phone .  If you didn't do this you will get the error like described above. https://support.google.com/pay/answer/7625139?hl=en&co=GENIE.Platform%3DAndroid

**Reply by Gopal6696** [reply]
i had configured googlepay with my real account

**Reply by Gopal6696** [reply]
@remonh87 im from india, stripe account created for USA

is there any problem for this case?

**Comment by dineshMadvise** [comment]
I also get this error 

Error Details
Gpay Error: StripeException(error: LocalizedErrorMessage(code: FailureCode.Failed, localizedMessage: presentForPaymentIntent() may only be called when Google Pay is available on this device., message: presentForPaymentIntent() may only be called when Google Pay is available on this device., stripeErrorCode: null, declineCode: null, type: null)

**Reply by Gopal6696** [reply]
@dineshMadvise let me know if you find any solution for this

**Reply by dineshMadvise** [reply]
> @dineshMadvise let me know if you find any solution for this

no
